### DISCUSSION TITLE
kdenlive

### DISCUSSION THREAD

**Discussion by joskezelensky** [mifi/lossless-cut#2182] [Q&A]
Is it possible to convert/ open a kdenlive video editor (xml) file in losslesscut and apply the edits losslessly that way?
For example: the kdenlive file includes some cuts and audio fades in / out.

**Comment by mifi** [comment]
Kdenslive xml is not currently supported, but support could be added if the format is relatively compatible with losslesscut's operation.

I've added your suggestion here: #1340

audio fade in / out will not be able to be supported, see #372

**Reply by joskezelensky** [reply]
> audio fade in / out will not be able to be supported, see #372

That's completely normal, but that will also be able to be done losslessly in kdenlive, so no problem
and i'm amazed by how fast your answer was given!

**Reply by joskezelensky** [reply]
> Kdenslive xml is not currently supported, but support could be added if the format is relatively compatible with losslesscut's operation.
> 
> I've added your suggestion here: #1340
> 
> audio fade in / out will not be able to be supported, see #372

https://github.com/nitanmarcel/kdl2fcp/blob/master/kdl2fcp.py
this is a python script to convert kdenlive to fcpxml, which should be a temporary solution for me...

**Reply by mifi** [reply]
if you can provide simple kdenlive xml file that covers everything that needs to be supported, then maybe someone (or an AI) can implement a parser

**Reply by mifi** [reply]
openai created this code:

```js
export function parseKdenlive(xmlStr: string) {
  const parser = new XMLParser();
  const jsonObj = parser.parse(xmlStr);

  // Assuming the structure of the Kdenlive project XML
  const project = jsonObj.kdenlive;
  const tracks = project.tracks.track.map((track: any) => ({
    id: track['@_id'],
    name: track['@_name'],
    clips: track.clip.map((clip: any) => ({
      id: clip['@_id'],
      name: clip['@_name'],
      start: clip['@_start'],
      end: clip['@_end'],
    })),
  }));

  return {
    projectName: project['@_name'],
    tracks,
  };
}
```

**Reply by joskezelensky** [reply]
> openai created this code:
> 
> ```js
> export function parseKdenlive(xmlStr: string) {
>   const parser = new XMLParser();
>   const jsonObj = parser.parse(xmlStr);
> 
>   // Assuming the structure of the Kdenlive project XML
>   const project = jsonObj.kdenlive;
>   const tracks = project.tracks.track.map((track: any) => ({
>     id: track['@_id'],
>     name: track['@_name'],
>     clips: track.clip.map((clip: any) => ({
>       id: clip['@_id'],
>       name: clip['@_name'],
>       start: clip['@_start'],
>       end: clip['@_end'],
>     })),
>   }));
> 
>   return {
>     projectName: project['@_name'],
>     tracks,
>   };
> }
> ```

It might be time to implement multi-input support
